packages/mask/src/plugins/IdeaMarket/SNSAdaptor/ListingsView.tsx

```diff
@@ -0,0 +1,364 @@
+            headerAlign: 'center' as const,
+            width: 95,
+            type: 'number',
+            align: 'center' as const,
+            valueFormatter: (params: GridValueFormatterParams) => formatterToUSD.format(params.value as number),
+        },
+        {
+            field: 'button',
+            headerName: '',
+            sortable: false,
+            width: 80,
+            align: 'center' as const,
+            renderCell: (params: GridRenderCellParams) => (
+                <Grid container alignContent="center" justifyContent="center">
+                    <Grid item>
+                        <Button
+                            href={`${BASE_URL}/i/${params.row.id}`}
+                            target="_blank"
+                            color="primary"
+                            size="small"
+                            variant="contained">
+                            {t('plugin_ideamarket_buy')}
+                        </Button>
+                    </Grid>
+                </Grid>
+            ),
+        },
+    ]
+
+    function CustomLoading() {
+        return (
+            <div className={classes.empty}>
+                <LoadingAnimation />
+            </div>
+        )
+    }
+
+    function NoRowsCustomOverlay() {
+        return (
+            <Stack height="100%" alignItems="center" justifyContent="center">
+                {t('no_data')}
+            </Stack>
+        )
+    }
+
+    function NoResultsCustomOverlay() {
+        return (
+            <Stack height="100%" alignItems="center" justifyContent="center">
+                {t('no_data')}
+            </Stack>
+        )
+    }
+
+    function handlePageChange(newPage: number) {
+        setPage(newPage)
+    }
+
+    return (
+        <div className={classes.root}>
+            <DataGrid
+                className={classes.grid}
+                paginationMode="server"
+                rowsPerPageOptions={[20]}
+                page={page}
+                pageSize={20}
+                onPageChange={handlePageChange}
+                headerHeight={18}
+                components={{
+                    Toolbar: QuickSearchToolbar,
+                    LoadingOverlay: CustomLoading,
+                    NoResultsOverlay: NoResultsCustomOverlay,
+                    NoRowsOverlay: NoRowsCustomOverlay,
+                }}
+                rowHeight={60}
+                rows={tokensFormatted ?? EMPTY_LIST}
+                rowCount={1000}
+                loading={loading}
+                columns={columns}
+                componentsProps={{
+                    toolbar: {
+                        value: searchText,
+                        onChange: (event: React.ChangeEvent<HTMLInputElement>) => {
+                            setPage(0)
+                            setSearchText(event.target.value)
+                        },
+                        clearSearch: () => setSearchText(''),
+                        filters: filters,
```

Comment:
@SebastianLF Take a look at these warnings. We've just updated the lint rules. It prefers property shorthand syntax.

Reply:
Done